07-http_gordle/6_6_query_param/internal/handlers/newgame/handler.go

```diff
@@ -0,0 +1,76 @@
+package newgame
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/oklog/ulid/v2"
+
+	"learngo-pockets/httpgordle/internal/api"
+	"learngo-pockets/httpgordle/internal/gordle"
+	"learngo-pockets/httpgordle/internal/session"
+)
+
+type gameAdder interface {
+	Add(game session.Game) error
+}
+
+// Handle returns the handler for the game creation endpoint.
+func Handle(db gameAdder) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		lang := r.URL.Query().Get(api.Lang)
+		if len(lang) > 0 {
+			// TODO create a game in the chosen language
+			fmt.Println(lang)
+		}
+
+		game, err := createGame(db)
+		if err != nil {
+			log.Printf("unable to create a new game: %s", err)
+			http.Error(w, "failed to create a new game", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+
+		apiGame := api.ToGameResponse(game)
+		err = json.NewEncoder(w).Encode(apiGame)
+		if err != nil {
+			http.Error(w, "failed to write response", http.StatusInternalServerError)
+		}
+	}
+}
+
+const maxAttempts = 5
+
+var corpusPath = "corpus/english.txt"
+
+func createGame(db gameAdder) (session.Game, error) {
```

Comment:
The name of the gameAdder variable is not clear enough. I'd name it "ga" here.

Reply:
adder instead of ga